test/iterator.js

```diff
@@ -643,6 +643,99 @@ metatests.testSync('Iterator.chainApply on string', test => {
   test.strictSame(actual, 'h e l l o');
 });
 
+metatests.testSync('Iterator.max no values', test => {
+  const actual = iter([]).max();
+  test.strictSame(actual, undefined);
+});
+
+metatests.testSync('Iterator.max simple', test => {
+  const actual = iter([1, 2, 3, 4, 5]).max();
+  test.strictSame(actual, 5);
+});
+
+metatests.testSync('Iterator.max accessor', test => {
+  const arr = [{ val: 5 }, { val: 4 }, { val: 3 }, { val: 9 }, { val: 4 }];
+  const actual = iter(arr).max(v => v.val);
+  test.strictSame(actual, arr[3]);
+});
+
+metatests.testSync('Iterator.max accessor this', test => {
+  const arr = [5, 4, 3, 2, 1];
+  const actual = iter(arr).max(
+    function(v) {
+      const val = this.curr;
+      this.curr = v;
+      return val;
+    },
+    { curr: 0 }
+  );
+  test.strictSame(actual, 4);
+});
+
+metatests.testSync('Iterator.min no values', test => {
+  const actual = iter([]).min();
+  test.strictSame(actual, undefined);
+});
+
+metatests.testSync('Iterator.min simple', test => {
+  const actual = iter([1, 2, 3, 4, 5]).min();
+  test.strictSame(actual, 1);
+});
+
+metatests.testSync('Iterator.min accessor', test => {
+  const arr = [{ val: 5 }, { val: 4 }, { val: 3 }, { val: 9 }, { val: 4 }];
+  const actual = iter(arr).min(v => v.val);
+  test.strictSame(actual, arr[2]);
+});
+
+metatests.testSync('Iterator.min accessor this', test => {
+  const arr = [1, 2, 3, 4, 5];
+  const actual = iter(arr).min(
+    function(v) {
+      const val = this.curr;
+      this.curr = v;
+      return val;
+    },
+    { curr: 10 }
+  );
+  test.strictSame(actual, 2);
```

Comment:
Same question here, I don't understand, how did you get 2 instead of 1?

Reply:
The order is like this
1. value: 1, curr: 10, accessor returned 10
2. value: 2, curr: 1, accessor returned 1 **our min value**
3. value: 3, curr: 2, accessor returned 2
4. value: 4, curr: 3, accessor returned 3
5. value: 5, curr: 4, accessor returned 4